### PR TITLE
grappling gun

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
adds a new weapon to mining, the grapple gun. this weapon is a new fun way to traverse lavaland, it lets u grapple onto things and ziplines u towards them. 

https://github.com/tgstation/tgstation/assets/138636438/5e935776-1482-4994-bcc7-e1ca7fcf33dc

while its less practical than things like the spinning mining modsuit module, or xenobio wings, its still very fun to use. 
u can also use this as a weapon against mobs! (like batman) 

the zipline will launch u towards the mob, doing some damage to them and launching them back

https://github.com/tgstation/tgstation/assets/138636438/3bf77fe5-4ed1-4153-8673-a0786ecee98f

## Why It's Good For The Game
adds a new fun harmless way to travel through mining biomes 

## Changelog
:cl:
add: miners can now purchase grapple guns
/:cl:

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
Common ben10 W

**Comment by MrMelbert** [comment]
more ways to grief lavaland syndicates I see

**Comment by GuillaumePrata** [comment]
Now can we grapple to the floors above Tenchu style?

File: code/datums/beam.dm [hunk 1]
```diff
@@ -278,7 +283,7 @@
...
  * beam_type: The type of your custom beam. This is for adding other wacky stuff for your beam only. Most likely, you won't (and shouldn't) change it.
  */
-/atom/proc/Beam(atom/BeamTarget,icon_state="b_beam",icon='icons/effects/beam.dmi',time=INFINITY,maxdistance=INFINITY,beam_type=/obj/effect/ebeam, beam_color = null, emissive = TRUE, override_origin_pixel_x = null, override_origin_pixel_y = null, override_target_pixel_x = null, override_target_pixel_y = null)
-	var/datum/beam/newbeam = new(src,BeamTarget,icon,icon_state,time,maxdistance,beam_type, beam_color, emissive, override_origin_pixel_x, override_origin_pixel_y, override_target_pixel_x, override_target_pixel_y )
+/atom/proc/Beam(atom/BeamTarget,icon_state="b_beam",icon='icons/effects/beam.dmi',time=INFINITY,maxdistance=INFINITY,beam_type=/obj/effect/ebeam, beam_color = null, emissive = TRUE, override_origin_pixel_x = null, override_origin_pixel_y = null, override_target_pixel_x = null, override_target_pixel_y = null, layer = ABOVE_ALL_MOB_LAYER)
```
Comment (san7890) [inline code/datums/beam.dm:286]: multiline this it's gotten hard to read

**Comment by Xackii** [comment]
It would be cool if you could use a hook to climb to z-levels above

**Comment by EricZilla** [comment]
I was gonna say yeah, having this work as a climbing rope when used inhand would be a pretty cool use too, gives it a reason to exist on icebox

**Comment by Elli-Skala** [comment]
R.i.p. jump boots to the creep of power

**Comment by necromanceranne** [comment]
listen, nobody liked jump boots. Can't use them with the modsuit, can't reliably use it to evade attacks, and you basically only saw them used to propel yourself at extreme speeds in zero gravity.

Or, what I did once, which was use them with hulk to bullrush people, but I don't expect people to do that.

**Comment by Xackii** [comment]
Boots still allow you to quickly run away or rush towards an enemy without taking up a hand slot.

**Comment by Admiral-Gigaton** [comment]
based PR

**Comment by HoodMilk** [comment]
HE DOESN'T MISS

**Comment by deathrobotpunch** [comment]
why is ben100 not a maintainer yet?

**Comment by scrunklershrungle932** [comment]
does slamming into enemies raise the style meter if you have it, and does it work against creatures that aren't naturally on lavaland?

**Comment by carlarctg** [comment]
what if it had an alt mode to create permanent ziplines at the cost of the hook (you can buy more)

**Comment by carlarctg** [comment]
PR so good it got ported before being merged

**Comment by Ben10Omintrix** [comment]
> I was gonna say yeah, having this work as a climbing rope when used inhand would be a pretty cool use too, gives it a reason to exist on icebox

i think there is already a climbing rope item that serves this purpose

**Comment by Ben10Omintrix** [comment]
> does slamming into enemies raise the style meter if you have it, and does it work against creatures that aren't naturally on lavaland?

it works on any creatures, but this gun only works in low pressure areas

**Comment by Ben10Omintrix** [comment]
> what if it had an alt mode to create permanent ziplines at the cost of the hook (you can buy more)

thatd be a cool idea but id have to change so much of the implementation, i can do it as a future pr

**Comment by Xackii** [comment]
> > does slamming into enemies raise the style meter if you have it, and does it work against creatures that aren't naturally on lavaland?
> 
> it works on any creatures, but this gun only works in low pressure areas

make it possible to use in all pressure if emaged

**Comment by Admiral-Gigaton** [comment]
> > > does slamming into enemies raise the style meter if you have it, and does it work against creatures that aren't naturally on lavaland?
> > 
> > 
> > it works on any creatures, but this gun only works in low pressure areas
> 
> make it possible to use in all pressure if emaged

funny, but like megaphone emagging, no one will realise or use it, likely, due to the rarity of a miner traitor actually using miner gear like that, and such.

something like hacking it, to make it do less damage or move slower, but allow in pressure? or some upgrade, whatever

**Comment by Xackii** [comment]
> > > > does slamming into enemies raise the style meter if you have it, and does it work against creatures that aren't naturally on lavaland?
> > > 
> > > 
> > > it works on any creatures, but this gun only works in low pressure areas
> > 
> > 
> > make it possible to use in all pressure if emaged
> 
> funny, but like megaphone emagging, no one will realise or use it, likely, due to the rarity of a miner traitor actually using miner gear like that, and such.
> 
> something like hacking it, to make it do less damage or move slower, but allow in pressure? or some upgrade, whatever

Well, the plus is that the grappling gun can be bought for any role if you just dig a mine or ask a miner to buy it for you. And the grappling gun itself looks very fun, and if add some kind of opportunity to use it at the station, it will be cool

**Comment by Elli-Skala** [comment]
Wait what if we applied this to a mod module instead (to replace the teather)

**Comment by Ben10Omintrix** [comment]
Oop don't merge this yet. There's an important edge case I forgot to test and I can't set it on draft cause I'm on mobile

**Pull request merged**